src/db/collections/application_state.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use mongodb::{bson::doc, error::Error, options::UpdateOptions};
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    db::{
+        mongodb::{MongoDbCollection, MongoDbCollectionExt},
+        MongoDb,
+    },
+    types::tangle::MilestoneIndex,
+};
+
+/// The MongoDb document representation of singleton Application State.
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct ApplicationStateDocument {
+    pub starting_index: MilestoneIndex,
```

Comment:
Maybe we should at the `MilestoneTimestamp` as well if we want to use the state in other areas as well?

Reply:
So far we have no use case for that, do we?

Reply:
It would still be nice for debugging purposes even if we don't expose it yet.

Reply:
Alright, I'll add it.